### PR TITLE
Fix video window title for Windows

### LINKED ISSUE #862: video window has title "TEMP TILTE CHANGE ME"
on windows: video window has title "TEMP TITLE CHANGE ME"

please fix

### PR DESCRIPTION
**Pull request by thorpelawrence**
Fixes #862

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/882)

### CODE REVIEW

File: src/windows/window.c [hunk 1]
```diff
@@ -77,7 +78,15 @@ UTOX_WINDOW *native_window_create_main(int x, int y, int w, int h) {
 
 HWND native_window_create_video(int x, int y, int w, int h) {
     LOG_DEBUG("Windows WM", "Creating video window");
-    HWND win = CreateWindowExW(0, L"uTox", L"TEMP TITLE CHANGE ME", WS_OVERLAPPEDWINDOW,
+
+    STRING *s = SPTR(WINDOW_TITLE_VIDEO_PREVIEW);
+    char pretitle[128];
+    snprintf(pretitle, 128, "%s", s->str);
```
Comment (GrayHatter) [inline src/windows/window.c]: Did you try `swprintf`?

**Comment by endoffile78** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by endoffile78** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882#-:-Kiq7bH-hm23pwBqgIy1:bnfp4nl)*

**Comment by robinlinden** [comment]
Thanks for fixing this! :lgtm_strong:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882#-:-KitTQeVQjjsa6NpmYTl:blhjsln)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 84 at r1](https://reviewable.io:443/reviews/utox/utox/882#-KipmOzgAj5XD6q31-QV-r1-84:-KitsC8YBD5S8tkrzsQk:b-rek47t) ([raw file](https://github.com/utox/utox/blob/a6cfc64648b71b0e2b5fccc3923ef5d967ee984d/src/windows/window.c#L84)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

Did you try `swprintf`?
</blockquote></details>

Fixed, now uses `swprintf`, thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by GrayHatter** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KivNKtyatiZiAl7DZVf:b-a0830t) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
> ```C
> 
>     wchar_t title[128];
>     swprintf(title, 128, L"%S", S(WINDOW_TITLE_VIDEO_PREVIEW));
> ```

needs to be `L"%s"` not `L"%S"` it's a wchat taking a char. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-Kiyb6K5mLfahjUR4JV1:bykmfr9) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

needs to be `L"%s"` not `L"%S"` it's a wchat taking a char. 
</blockquote></details>

Tried with `%s` but didn't work, shouldn't `%S` be used with a char in swprintf as `%s` is for wchar?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by GrayHatter** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEMOoSdExLZa-eUTMo:b-u7axbg) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

Tried with `%s` but didn't work, shouldn't `%S` be used with a char in swprintf as `%s` is for wchar?
</blockquote></details>

I'm very sorry for the delay in reply, somehow this fell off my todo list... I didn't mean to leave you hanging for so long.

I just double checked and according to http://en.cppreference.com/w/cpp/io/c/fwprintf `%s` reads from a `char` string, where `%ls` reads from a wchar string. All of the i18n strings in utox are just char. see also https://linux.die.net/man/3/wprintf

man 3 snprintf also says `%S` is an alias for `%ls` Don't use. So no matter what, we shouldn't use `%S`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjETbO8OgHHaKW0NFnD:b9bn07y) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

I'm very sorry for the delay in reply, somehow this fell off my todo list... I didn't mean to leave you hanging for so long.

I just double checked and according to http://en.cppreference.com/w/cpp/io/c/fwprintf `%s` reads from a `char` string, where `%ls` reads from a wchar string. All of the i18n strings in utox are just char. see also https://linux.die.net/man/3/wprintf

man 3 snprintf also says `%S` is an alias for `%ls` Don't use. So no matter what, we shouldn't use `%S`
</blockquote></details>

No worries, so if they're all char should it just use `swprintf("%ls")` or try something different, or use the original `fwprintf`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by GrayHatter** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEZ3k6HD46zvSj5TXk:b-yj98oe) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

No worries, so if they're all char should it just use `swprintf("%ls")` or try something different, or use the original `fwprintf`?
</blockquote></details>

use `swprintf("%s")`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEdsb97K5i4T-v8wzp:b-pw4fnl) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

use `swprintf("%s")`
</blockquote></details>

That was the original issue I had with it displaying the wrong title, since as you say they are just char strings couldn't `fwprintf("%s")` be used?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by GrayHatter** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEfYXMD-6kcdhtxw9A:b-cz090e) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

That was the original issue I had with it displaying the wrong title, since as you say they are just char strings couldn't `fwprintf("%s")` be used?
</blockquote></details>

what's the problem with swprintf()?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEgqbJM4hSCqNcdgn8:bb7hm6r) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

what's the problem with swprintf()?
</blockquote></details>

With `swprintf("%s")` the characters are garbled 


---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEi_-cN6zmk8a26buE:bhyjwqk) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

With `swprintf("%s")` the characters are garbled 

</blockquote></details>

With the previous attempt it was showing 'Video preview'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by GrayHatter** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEjNcXhITRfvTAWk54:b-9u1n7j) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

With the previous attempt it was showing 'Video preview'
</blockquote></details>

if `%ls` works use that, and I'll look into what's wrong later, add a // TODO grayhatter as well so I'll remember this thread

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEm-klyqbv2F3cJwJu:b-tw412v) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

if `%ls` works use that, and I'll look into what's wrong later, add a // TODO grayhatter as well so I'll remember this thread
</blockquote></details>

Doesn't seem that `%ls` is an alias for `%S` with `swprintf`, so looks like I may have to use `swprintf` with `%S`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/windows/window.c, line 83 at r2](https://reviewable.io:443/reviews/utox/utox/882#-KivNKtxJc-gI1_0DyjQ:-KjEmM_ccET6cQjR4QBS:b-xbisd7) ([raw file](https://github.com/utox/utox/blob/2ef7e4986822379390ce119b3418f7de9421b4ca/src/windows/window.c#L83)):*
<details><summary><i>Previously, thorpelawrence (Lawrence Thorpe) wrote…</i></summary><blockquote>

Doesn't seem that `%ls` is an alias for `%S` with `swprintf`, so looks like I may have to use `swprintf` with `%S`
</blockquote></details>

`swprintf(title, 128, L"%ls", S(WINDOW_TITLE_VIDEO_PREVIEW));`
has a different result to
`swprintf(title, 128, L"%S", S(WINDOW_TITLE_VIDEO_PREVIEW));`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by robinlinden** [comment]
It's a Microsoft thing.
https://msdn.microsoft.com/en-us/library/aa272865(v=vs.60).aspx
"The types C and S, and the behavior of  c and s with printf functions, are Microsoft extensions and are not ANSI-compatible."
`%s` in `wprintf`-functions means it's using a wide string. `%S` means it's using a single-byte character string.

**Comment by GrayHatter** [comment]
Windows quality API there...

**Comment by thorpelawrence** [comment]
That's where I originally saw the %S behavior, should work now

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882#-:-KjEuOZW9fciyqisvAwz:be5v6lo)*

**Comment by GrayHatter** [comment]
My apologies again, if you ever have API docs to back up the decisions you've made, link them and I'll shutup :grin: 

:lgtm: 

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882#-:-KjEvyJRdxLLm_QypRHT:b363peo)*

**Comment by robinlinden** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882)*

**Comment by thorpelawrence** [comment]
No problem, strange that Microsoft does it differently so was scratching my head there for a bit

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/882#-:-KjEwR3GWr5KAa_kUM9S:b-v24j44)*

**Comment by GrayHatter** [comment]
@thorpelawrence thanks mate it's fixed now!

**Pull request merged**